js/screens/onboarding/PinScreen.js

```diff
@@ -6,31 +6,296 @@ import {
+      this.pinConfirmComponent.clear()
+    }
+    this.setState({
+      pinState: {
+        state: 'PinUnselected'
+      }
+    })
+  }
+
+  // Dispatch the Action that save the PIn in the Keychain
+  createPin = (pin: string) => {
+    this.props.dispatch(createPin(pin))
+  }
+
+  // Render a different header when the user need to confirm the PIN
+  renderContentHeader = (pinState: PinState): React.Node => {
+    if (pinState.state === 'PinUnselected') {
+      return <H1>{I18n.t('onboarding.pin.contentTitle')}</H1>
+    } else {
+      return <H1>{I18n.t('onboarding.pin.contentTitleConfirm')}</H1>
+    }
+  }
+
+  // Render the PIN match/doesn't match feedback message
+  renderCodeInputConfirmValidation = (pinState: PinConfirmed): React.Node => {
+    const validationMessage = pinState.hasError ? (
+      <TextWithIcon error>
+        <Icon name={'cross'} />
+        <Text>{I18n.t('onboarding.pin.confirmInvalid')}</Text>
+      </TextWithIcon>
+    ) : (
+      <TextWithIcon success>
+        <Icon name={'check'} />
+        <Text>{I18n.t('onboarding.pin.confirmValid')}</Text>
+      </TextWithIcon>
+    )
+    return (
+      <React.Fragment>
+        <View spacer extralarge />
+        {validationMessage}
+      </React.Fragment>
+    )
+  }
+
+  // Render select/confirm CodeInput component
+  renderCodeInput = (pinState: PinState): React.Node => {
+    if (pinState.state === 'PinUnselected') {
+      /**
+       * The CodeInput component where the user SELECT the PIN.
+       *
+       * autofucus={false}
+       * onFulfill={this.onPinFulfill}
+       */
+      return (
+        <CodeInput
+          ref={(el: React.ElementRef<typeof CodeInput>): void =>
+            (this.pinComponent = el)
+          }
+          secureTextEntry
+          keyboardType="numeric"
+          autoFocus={false}
+          className="border-b"
+          codeLength={5}
+          cellBorderWidth={2}
+          inactiveColor={'rgba(92,111,130,0.5)'}
+          activeColor={'rgba(23,50,77,1)'}
+          onFulfill={this.onPinFulfill}
+          codeInputStyle={{ fontSize: 24, height: 56 }}
+        />
+      )
+    } else {
+      /**
+       * The CodeInput component where the user CONFIRM the PIN.
+       *
+       * autofucus={true}
+       * compareWithCode={pinState.pin}
+       * onFulfill={this.onPinConfirmFulfill}
+       * */
+      return (
+        <React.Fragment>
+          <CodeInput
+            ref={(el: React.ElementRef<typeof CodeInput>): void =>
+              (this.pinConfirmComponent = el)
+            }
+            secureTextEntry
+            keyboardType="numeric"
+            autoFocus
+            compareWithCode={pinState.pin}
+            className="border-b"
+            codeLength={5}
+            cellBorderWidth={2}
+            inactiveColor={'rgba(92,111,130,0.5)'}
+            activeColor={'rgba(23,50,77,1)'}
+            onFulfill={this.onPinConfirmFulfill}
+            codeInputStyle={{ fontSize: 24, height: 56 }}
```

Comment:
with another PR, can we have these fontSize taken from a predefined set of constants ? alike
https://github.com/jxnblk/styled-system

Reply:
I think we can change this also in this PR.